### DISCUSSION TITLE
What powers this wonderful thing?

### DISCUSSION THREAD

**Discussion by githubUser01946** [jesusjimsa/beatles-lyrics-every-hour#12] [Q&A] [answered]
Its all in the title, but is it hosted on google or something else?

**Comment by jesusjimsa** [comment, marked as answer]
Hi @githubUser01946!

It is done here directly, with GitHub Actions. I set up an Action that runs every hour, and this Action just launches the script that selects a line from the file and posts it on Twitter and Mastodon.

This is the Action: https://github.com/jesusjimsa/beatles-lyrics-every-hour/blob/master/.github/workflows/send_tweet.yml

There is also a limitation from GitHub that disables Actions on repositories that haven't been active for a month (no commits). So I also set up an action that creates a dummy commit and pushes it. It just writes the current date once a month to keep everything running.

This is the one for dummy commits: https://github.com/jesusjimsa/beatles-lyrics-every-hour/blob/master/.github/workflows/update_repo.yml

**Comment by githubUser01946** [comment]
Ok but how do you get it from here to mastodon?

**Reply by jesusjimsa** [reply]
For Mastodon, I use an API called Mastodon.py: https://github.com/halcy/Mastodon.py

It's very easy, I had it ready pretty fast.

**Comment by githubUser01946** [comment]
Cool, thanks!

Is there a tutorial somewhere so I can get something up and running?

**Reply by jesusjimsa** [reply]
This is the one I used: https://martinheinz.dev/blog/86

**Reply by githubUser01946** [reply]
I will check it out and thanks for sharing the knowledge
